include/ginkgo/core/base/mpi.hpp

```diff
@@ -1166,7 +1320,54 @@ class window {
         GKO_ASSERT_NO_MPI_ERRORS(MPI_Rput(
             origin_buffer, origin_count, type_impl<PutType>::get_type(),
             target_rank, target_disp, target_count,
-            type_impl<PutType>::get_type(), get_window(), req.get()));
+            type_impl<PutType>::get_type(), this->get_window(), req.get()));
+        return req;
+    }
+
+    /**
+     * Accumulate data into the target window.
+     *
+     * @param origin_buffer  the buffer to send
+     * @param origin_count  the number of elements to put
+     * @param target_rank  the rank to put the data to
+     * @param target_disp  the displacement at the target window
+     * @param target_count  the request handle for the send call
+     * @param operation  the reduce operation. See @MPI_Op
+     */
+    template <typename PutType>
+    void accumulate(const PutType* origin_buffer, const int origin_count,
+                    const int target_rank, const unsigned int target_disp,
+                    const int target_count, MPI_Op operation) const
+    {
+        GKO_ASSERT_NO_MPI_ERRORS(MPI_Accumulate(
+            origin_buffer, origin_count, type_impl<PutType>::get_type(),
+            target_rank, target_disp, target_count,
+            type_impl<PutType>::get_type(), operation, this->get_window()));
+    }
+
+    /**
+     * (Non-blocking) Accumulate data into the target window.
+     *
+     * @param origin_buffer  the buffer to send
+     * @param origin_count  the number of elements to put
+     * @param target_rank  the rank to put the data to
+     * @param target_disp  the displacement at the target window
+     * @param target_count  the request handle for the send call
+     * @param operation  the reduce operation. See @MPI_Op
+     *
+     * @return  the request handle for the send call
+     */
+    template <typename PutType>
+    request r_accumulate(const PutType* origin_buffer, const int origin_count,
```

Comment:
Test for this one?

Reply:
It should be there. See [Line 349](https://github.com/ginkgo-project/ginkgo/blob/mpi-fixes/core/test/mpi/base/bindings.cpp#L349)